packages/analytics-browser/src/storage/local-storage.ts

```diff
@@ -1,8 +1,21 @@
 import { getGlobalScope } from '@amplitude/analytics-client-common';
 import { BrowserStorage } from './browser-storage';
+import { Logger } from '@amplitude/analytics-types';
 
 export class LocalStorage<T> extends BrowserStorage<T> {
-  constructor() {
-    super(getGlobalScope()?.localStorage);
+  logger?: Logger;
+
+  constructor(logger?: Logger) {
```

Comment:
i think we want the param to be an object with `loggerProvider` as a key. that way we can extend this to be of type `BrowserOptions` (not to be mistaken with `BrowserConfig`) if needed in the future

Reply:
I make an interface called `localStorageOptions` with logger as an property and use this interface as the type of the param of local storage constructor.